src/components/Transactions.js

```diff
@@ -27,15 +32,28 @@ class Transactions extends React.Component {
       loaded: false,
       hasAfter: false,
       hasBefore: false,
+      queryParams: this.obtainQueryParams(),
     }
   }
 
   componentDidMount() {
-    this.getData(true, null, null, '');
+    this.getData(this.state.queryParams);
 
     WebSocketHandler.on('network', this.handleWebsocket);
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    const queryParams = this.obtainQueryParams();
+
+    // Do we have new URL params?
+    if (JSON.stringify(this.state.queryParams) !== JSON.stringify(queryParams)) {
```

Comment:
Even though `JSON.stringify` usually works, it may fail because of the order the keys are created. In this case, we can either add a new dependency (such as `loadsh`) or create our own shallow comparison method. What do you think?

Reply:
We already have `lodash` on `wallet-lib` so it wouldn't be a new dependency to be installed. We could add it to this project and use.